# QTM 385

***

## Homework 1

Student: Yue Gao Leah [2323100]


***

### Question 1

Run the following code:

```
0.1 + 0.1 + 0.1
```

Is it equal to `0.3`? If not, why?

In [1]:
# Your answer here
0.1 + 0.1 +0.1 

0.30000000000000004

In [3]:
0.1 + 0.1 + 0.1 == 0.3

False

It is not equal to 0.3. Computers use binary code instead of base 10 system. Since 0.1 can't be mathmatically expressed with base 2 system, adding three "0.1" together will give us a number that is very close but not equal to 0.3.


***

### Question 2

How many digits has $3^{1,000,000}$? (Hint: the function `str` transforms a number into a string.)

In [9]:
# Your answer here
len(str(3**1000000))

477122

$3^{1,000,000}$ has 477122 digits.


***

### Question 3

Create a tuple with three foods that you love to eat. Save it in a variable called `preferred_food`.

In [10]:
# Your answer here
preferred_food = ("mango", "hotpot", "sushi")


***

### Question 4

The module random has a handy function for selecting elements in a tuple: `choice`. Import the module `random`, and name it `rd`.

In [11]:
# Your answer here
import random as rd


***

### Question 5

Using the function choice, select a food to eat today. (Hint: when we used `numpy`, I named it `np`. When I needed a function in it, I did `np.function_I_needed`...)

In [12]:
# Your answer here
rd.choice(preferred_food)

'sushi'